addons/sale_stock_margin/models/sale_order_line.py

```diff
@@ -13,6 +13,6 @@ def _compute_purchase_price(self):
         for line in self:
             if not line.move_ids:
                 lines_without_moves |= line
-            else:
+            elif line.product_id.categ_id.property_cost_method != 'standard':
```

Comment:
IMO, This condition should be in the first `if` (inversed of course). 
Because, in this current version, if there is sale_stock_margin installed, and `line.product_id.categ_id.property_cost_method == 'standard' and line.move_ids`, we won't call the super of the method.
Then, the `lines_without_moves` (maybe renamed the variables?) should also contains the case of `line.product_id.categ_id.property_cost_method == 'standard'`.
What do you think ?

Reply:
Indeed, in the current version the super of the method is never called. But if I apply the changes you mentioned, my problem will stay the same, the purchase price will be changed with the standard price of the product in the super of the method! Maybe we need to check if the price has changed in the order line before assigning the standard price of the product to the "purchase_price" attribute. What do you think ?

Reply:
Yeah my bad, Damn! Compute stored are painful sometime. 

>   need to check if the price has changed in the order line before assigning the standard price of the product to the "purchase_price" attribute
I don't see how you can do that. 

Then , IMO, Your fix seems legit. because it there are moves, we cannot modify the depends fields of  the super compute `'product_id', 'company_id', 'currency_id', 'product_uom'` in the sale line (then don't call the super shouldn't be a issue).

Reply:
Yes I agree with your explanation. Indeed, we don't need to call the super method in this case. The first solution seems correct to me also! Thanks for review